src/plugins/platforms/eglfs/qeglfscontext.cpp

```diff
@@ -48,15 +48,212 @@
+        for (int i = 0; i < STATE_GUARD_VERTEX_ATTRIB_COUNT; ++i) {
+            glFuncs.glGetVertexAttribiv(i, GL_VERTEX_ATTRIB_ARRAY_ENABLED, (GLint *) &m_vertexAttribs[i].enabled);
+            glFuncs.glGetVertexAttribiv(i, GL_VERTEX_ATTRIB_ARRAY_BUFFER_BINDING, (GLint *) &m_vertexAttribs[i].arrayBuffer);
+            glFuncs.glGetVertexAttribiv(i, GL_VERTEX_ATTRIB_ARRAY_SIZE, &m_vertexAttribs[i].size);
+            glFuncs.glGetVertexAttribiv(i, GL_VERTEX_ATTRIB_ARRAY_STRIDE, &m_vertexAttribs[i].stride);
+            glFuncs.glGetVertexAttribiv(i, GL_VERTEX_ATTRIB_ARRAY_TYPE, (GLint *) &m_vertexAttribs[i].type);
+            glFuncs.glGetVertexAttribiv(i, GL_VERTEX_ATTRIB_ARRAY_NORMALIZED, (GLint *) &m_vertexAttribs[i].normalized);
+            glFuncs.glGetVertexAttribPointerv(i, GL_VERTEX_ATTRIB_ARRAY_POINTER, &m_vertexAttribs[i].pointer);
+        }
+        glGetTexParameteriv(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, (GLint *) &m_minFilter);
+        glGetTexParameteriv(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, (GLint *) &m_magFilter);
+        glGetTexParameteriv(GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, (GLint *) &m_wrapS);
+        glGetTexParameteriv(GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, (GLint *) &m_wrapT);
+    }
+
+    ~StateGuard() {
+        QOpenGLFunctions glFuncs(QOpenGLContext::currentContext());
+
+        glFuncs.glUseProgram(m_program);
+        glActiveTexture(m_activeTextureUnit);
+        glBindTexture(GL_TEXTURE_2D, m_texture);
+        glFuncs.glBindFramebuffer(GL_FRAMEBUFFER, m_fbo);
+        glViewport(m_viewport[0], m_viewport[1], m_viewport[2], m_viewport[3]);
+        glDepthMask(m_depthWriteMask);
+        glColorMask(m_colorWriteMask[0], m_colorWriteMask[1], m_colorWriteMask[2], m_colorWriteMask[3]);
+        if (m_blend)
+            glEnable(GL_BLEND);
+        if (m_depth)
+            glEnable(GL_DEPTH_TEST);
+        if (m_cull)
+            glEnable(GL_CULL_FACE);
+        if (m_scissor)
+            glEnable(GL_SCISSOR_TEST);
+        for (int i = 0; i < STATE_GUARD_VERTEX_ATTRIB_COUNT; ++i) {
+            if (m_vertexAttribs[i].enabled)
+                glFuncs.glEnableVertexAttribArray(i);
+            GLuint prevBuf;
+            glGetIntegerv(GL_ARRAY_BUFFER_BINDING, (GLint *) &prevBuf);
+            glFuncs.glBindBuffer(GL_ARRAY_BUFFER, m_vertexAttribs[i].arrayBuffer);
+            glFuncs.glVertexAttribPointer(i, m_vertexAttribs[i].size, m_vertexAttribs[i].type,
+                                          m_vertexAttribs[i].normalized, m_vertexAttribs[i].stride,
+                                          m_vertexAttribs[i].pointer);
+            glFuncs.glBindBuffer(GL_ARRAY_BUFFER, prevBuf);
+        }
+        glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, m_minFilter);
+        glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, m_magFilter);
+        glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, m_wrapS);
+        glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, m_wrapT);
```

Comment:
In the blit function, you are only setting MIN and MAG filter. You donæt touch the wrap modes, so you don't need to restore them either.